packages/front-end/components/Settings/EditDataSource/FeatureEvaluationQueries/FeatureEvaluationQueries.tsx

```diff
@@ -0,0 +1,273 @@
+import React, { FC, useCallback, useMemo, useState } from "react";
+import {
+  DataSourceInterfaceWithParams,
+  FeatureUsageQuery,
+} from "back-end/types/datasource";
+import cloneDeep from "lodash/cloneDeep";
+import { FaChevronRight, FaPlus } from "react-icons/fa";
+import { Box, Card, Flex, Heading } from "@radix-ui/themes";
+import { DataSourceQueryEditingModalBaseProps } from "@/components/Settings/EditDataSource/types";
+import DeleteButton from "@/components/DeleteButton/DeleteButton";
+import Code from "@/components/SyntaxHighlighting/Code";
+import MoreMenu from "@/components/Dropdown/MoreMenu";
+import Button from "@/ui/Button";
+import usePermissionsUtil from "@/hooks/usePermissionsUtils";
+import Callout from "@/ui/Callout";
+import { FeatureEvaluationQueryModal } from "./FeatureEvaluationQueryModal";
+
+type FeatureEvaluationQueriesProps = DataSourceQueryEditingModalBaseProps;
+type UIMode = "view" | "edit" | "add" | "dimension";
+export const FeatureEvaluationQueries: FC<FeatureEvaluationQueriesProps> = ({
+  dataSource,
+  onSave,
+  onCancel,
+  canEdit = true,
+}) => {
+  const intitialOpenIndexes: boolean[] = Array.from(
+    Array(dataSource.settings?.queries?.featureUsage?.length || 0),
+  ).fill(true);
+
+  const [uiMode, setUiMode] = useState<UIMode>("view");
+  const [editingIndex, setEditingIndex] = useState<number>(-1);
+  const [openIndexes, setOpenIndexes] =
+    useState<boolean[]>(intitialOpenIndexes);
+
+  const permissionsUtil = usePermissionsUtil();
+  canEdit = canEdit && permissionsUtil.canUpdateDataSourceSettings(dataSource);
+
+  const handleExpandCollapseForIndex = useCallback(
+    (index) => () => {
+      const currentValue = openIndexes[index] || false;
+      const updatedOpenIndexes = [...openIndexes];
+      updatedOpenIndexes[index] = !currentValue;
+
+      setOpenIndexes(updatedOpenIndexes);
+    },
+    [openIndexes],
+  );
+
+  const handleCancel = useCallback(() => {
+    setUiMode("view");
+    setEditingIndex(-1);
+    onCancel();
+  }, [onCancel]);
+
+  const featureUsageQueries = useMemo(
+    () => dataSource.settings?.queries?.featureUsage || [],
+    [dataSource.settings?.queries?.featureUsage],
+  );
+
+  console.log("featureUsageQueries", featureUsageQueries);
+
+  const handleAdd = useCallback(() => {
+    setUiMode("add");
+    setEditingIndex(featureUsageQueries.length);
+  }, [featureUsageQueries]);
+
+  const handleActionClicked = useCallback(
+    (idx: number, uiMode: UIMode) => async () => {
+      setEditingIndex(idx);
+      setUiMode(uiMode);
+    },
+    [],
+  );
+
+  const handleActionDeleteClicked = useCallback(
+    (idx: number) => async () => {
+      const copy = cloneDeep<DataSourceInterfaceWithParams>(dataSource);
+
+      // @ts-expect-error TS(2532) If you come across this, please fix it!: Object is possibly 'undefined'.
```

Comment:
issue: Can we fix this error & the other `ts-expect-errors` on this file while we are here?